keps/0007-diagnostic-template.md

```diff
@@ -0,0 +1,43 @@
+---
+kep-number: 7
+title: Diagnostic Template
+authors:
+  - "@gerred"
+  - "@meichstedt"
+owners:
+  - "@gerred"
+  - "@meichstedt"
+creation-date: 2019-04-11
+last-updated: 2019-04-11
+status: provisional
+---
+
+# KUDO Diagnostic Template
+
+- [KUDO Diagnostic Template](#kudo-diagnostic-template)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+
+## Summary
+
+The KUDO Diagnostic Template is intended to provide an opinionated template for issues when using frameworks for KUDO and creating frameworks with KUDO. This is intended to supersede `ISSUE_TEMPLATE.md` and provide an opinionated set of diagnostic information, logs, and tooling required to adequately diagnose and respond to issues with KUDO.
+
+## Motivation
+
+Engineering teams have many incoming priorities on top of work defined upfront during sprint planning. All time spent triaging issues takes away from estimating, prioritizing, and fixing issues. This KEP provides a framework for quickly triaging and identifying issues within KUDO.
+
+### Goals
+
+* Define criteria for submitting KUDO issues
+* Define tooling, logs, and metrics needed to support, diagnose, and fix issues reported
+
+### Non-Goals
+
+* Define frameworks for supporting, diagnosing, and fixing frameworks built using KUDO
+
+## Proposal
+
+TBD
```

Comment:
current template has:
1. what happened
2. what was expected
3. steps to reproduce
4. **Environment**:
- Kubernetes version (use `kubectl version`):
- Kudo version (use `kubectl kudoctl version`): 
- Framework:
- Frameworkversion:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

This is a good for step to bugs.    We need to define the min required to work on bugs.  I would put that as:
1. kudo operator and cli version
2. framework and version
3. kudo logs

This can be added and debated after we land this kep.